war/src/main/js/pluginSetupWizardGui.js

```diff
@@ -0,0 +1,555 @@
+				}
+			}
+			$('.match').parent().prev('h2').addClass('match');
+			$pl.addClass('searching');
+		}
+		else {
+			findIndex = 0;
+			$pl.removeClass('searching');
+		}
+		lastSearch = text;
+	};
+	
+	// handle input for the search here
+	$wizard.on('keyup change', '.plugin-select-controls input[name=searchbox]', function(e) {
+		var val = $(this).val();
+		searchForPlugins(val, true);
+	});
+	
+	// handle clearing the search
+	$wizard.on('click', '.clear-search', function() {
+		$('input[name=searchbox]').val('');
+		searchForPlugins('', false);
+	});
+	
+	// toggles showing the selected items as a simple search
+	var toggleSelectedSearch = function() {
+		var $srch = $('input[name=searchbox]');
+		var val = 'show:selected';
+		if($srch.val() == val) {
+			val = '';
+		}
+		$srch.val(val);
+		searchForPlugins(val, false);
+	};
+	
+	// handle clicking on the category
+	var selectCategory = function() {
+		$('input[name=searchbox]').val('');
+		searchForPlugins('', false);
+		var id = $(this).attr('href');
+		var $el = $($(this).attr('href'));
+		var $pl = $('.plugin-list');
+		var top = $pl.scrollTop() + $el.position().top;
+		$pl.stop(true).animate({
+			scrollTop: top
+		}, 250, function() {
+			var top = $pl.scrollTop() + $el.position().top;
+			$pl.stop(true).scrollTop(top);
+		})
+	};
+	
+	// handle show/hide details during the installation progress panel
+	var toggleInstallDetails = function() {
+		var $c = $('.install-console');
+		if($c.is(':visible')) {
+			$c.slideUp();
+		}
+		else {
+			$c.slideDown();
+		}
+	};
+	
+	// scoped click handler, prevents default actions automatically
+	var bindClickHandler = function(cls, action) {
+		$wizard.on('click', cls, function(e) {
+			action.apply(this, arguments);
+			e.preventDefault();
+		});
+	};
+	
+	// click action mappings
+	var actions = {
+		'.install-recommended': installDefaultPlugins,
+		'.install-custom': loadCustomPluginPanel,
+		'.install-home': function() { setPanel(welcomePanel); },
+		'.install-selected': function() { installPlugins(selectedPlugins); },
+		'.toggle-install-details': toggleInstallDetails,
+		'.install-done': finishInstallation,
+		'.plugin-select-all': function() { selectedPlugins = recommendedPlugins.slice(0); refreshPluginSelectionPanel(); },
+		'.plugin-select-none': function() { selectedPlugins = []; refreshPluginSelectionPanel(); },
+		'.plugin-select-recommended': function() { selectedPlugins = installData.defaultPlugins.slice(0); refreshPluginSelectionPanel(); },
+		'.plugin-show-selected': toggleSelectedSearch,
+		'.select-category': selectCategory
+	};
+	for(var cls in actions) {
+		bindClickHandler(cls, actions[cls]);
+	}
+	
+	// kick off to get resource bundle
+	jenkins.get('/i18n/resourceBundle?baseName=jenkins.install.pluginSetupWizard', function(res) {
+		if(res.status != 'ok') {
+			msg = { error: 'Unable to load localization data: ' + res.message };
+			setPanel(errorPanel);
+			return;
+		}
+		
+		msg = res.data;
```

Comment:
Can we call the localised messages something more meaningful (instead of `msg`)? E.g. `i18n`.